crates/pg_workspace_new/src/workspace/server.rs

```diff
@@ -342,6 +342,57 @@ impl Workspace for WorkspaceServer {
             skipped_diagnostics: 0,
         })
     }
+
+    #[tracing::instrument(level = "debug", skip(self))]
+    fn get_completions(
+        &self,
+        params: super::CompletionParams,
+    ) -> Result<pg_completions::CompletionResult, WorkspaceError> {
+        let doc = self
+            .documents
+            .get(&params.path)
+            .ok_or(WorkspaceError::not_found())?;
+
+        let offset = doc.line_and_col_to_offset(params.line, params.column);
```

Comment:
this should happen in the lsp layer. we already have helpers for that in `pg_lsp_converters`. the input to the workspace should only be `TextSize` or `TextRange`. the lsp server tracks documents too with the purpose of maintaining the `LineIndex`, which is the data structure we need for the conversion. another reason to handle it on lsp level are different encodings.

Reply:
good point, I didn't think about the encoding!